behind-the-veil-siteroot/imports/ui/components/pages/service/AddEditServicePage.jsx

```diff
@@ -14,26 +14,51 @@ import BackButton from "../../button/BackButton";
 import Input from "../../input/Input";
 import Button from "../../button/Button";
 
-import { CheckIcon, XMarkIcon, TrashIcon } from "@heroicons/react/24/outline";
+import {CheckIcon, TrashIcon, XMarkIcon} from "@heroicons/react/24/outline";
+import UrlBasePath from "../../../enums/UrlBasePath";
+import {Modal} from "react-responsive-modal";
 
-export const AddEditServicePage = ({ isEdit }) => {
+export const AddEditServicePage = ({isEdit}) => {
     const navigateTo = useNavigate();
 
+    // text for title/save button
     const title = isEdit ? "Edit Service" : "Add New Service";
     const button = isEdit ? "Edit Service" : "Add Service";
 
+    // input and error variables
     const [serviceName, setServiceName] = useState("");
     const [serviceType, setServiceType] = useState("None");
     const [serviceDuration, setServiceDuration] = useState(0);
     const [servicePrice, setServicePrice] = useState(0);
     const [serviceDescription, setServiceDescription] = useState("");
 
+    const [errors, setErrors] = useState({
+        serviceName: "",
+        serviceType: "",
+        serviceDuration: "",
+        servicePrice: "",
+        serviceDescription: "",
+        images: ""
+    })
+
+    // end status modal
+    const [open, setOpen] = useState(false);
+    const onOpenModal = () => setOpen(true);
+    const onCloseModal = () => setOpen(false);
+    const [isSuccess, setSuccess] = useState(false);
+
+    // get current user information
     const userInfo = useUserInfo();
+
+    // if user is not an artist, navigate them away
     if (userInfo.type !== "artist") {
         navigateTo(`/`);
     }
 
-    const serviceId = isEdit ? useLocation().pathname.split("/")[2] : "";
+    // get service ID from url
+    const {serviceId} = isEdit ? useParams() : "";
```

Comment:
Small thing, hooks (e.g. useParams) shouldn't be conditionally called, they should always be called at the top level of components, so that its guaranteed that they are called in the same order every render, which is just one of react's rules to maintain state correctly.

So I suggest doing something like:
`const params = useParams()
`const {serviceId} = isEdit ? params : ""

which does the same thing you want, except useParams is guaranteed to be called

Reply:
fixed! thank you!